prov/opx/configure.m4

```diff
@@ -121,6 +121,9 @@ AC_DEFUN([FI_OPX_CONFIGURE],[
 		    [],
 		    [opx_happy=0])
 
+		AS_IF([test $opx_happy -eq 1],[
+			AC_CHECK_DECL([HFI1_CAP_TID_RDMA], [], [opx_happy=0])
```

Comment:
```suggestion
						AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
				[[#include <rdma/hfi/hfi1_user.h>]],
				[[
					#if HFI1_USER_SWMAJOR != 6
					#error "incorrect version of hfi1_user.h"
					#endif
				]])],
				[AC_MSG_NOTICE([hfi1_user.h version 6.x found... yes])],
				[
				AC_MSG_NOTICE([hfi1_user.h version 6.x found... no])
				opx_happy=0
				])
```

Reply:
Can you try with that AC_COMPILE_IFELSE instead of the AC_CHECK_DECL and see if it skips the OPX build for your failing system? It depends on the HFI1_USER_SWMAJOR in that file.

If not, we'll have to check for the #define of HFI1_CAP_TID_RDMA inside there instead, but it should still work.